docs/tool/js/core/router.js

```diff
@@ -3,7 +3,16 @@
 var router = {
     isFilterInitialized: false,
     stateObj: {},
-    initFilters: function(msg,data){
+    initGate: function(){
+        if ( getState().initialProjectsRendered && getState().filterViewLoaded ) { // each stateChange triggers the gate function
```

Comment:
This is great. I was concerned about this when I was working on the "loading" screen because that projectsRendered event was just based on the slowness of mapbox not the sequencing of all the stuff needing to occur.

 We should think about whether it makes sense to mess around a bit more with the sequence and gate functionality so that there can be one master 'page done loading' event that is triggered - when router is needed, it's after router is done loading, but when router isn't needed it is this same gate conditions of what you have put here. Then the "loading" buttons in the modal can be based on that 'loading done' state event rather than the filterviewLoaded that I'm using now.